apps/arena/lib/arena/game_updater.ex

```diff
@@ -598,4 +610,72 @@ defmodule Arena.GameUpdater do
       _ -> entity_id
     end
   end
+
+  defp spawn_power_ups(
+         %{game_state: game_state, game_config: game_config} = state,
+         victim,
+         amount
+       )
+       when amount > 0 do
+    last_id = game_state.last_id + 1
+
+    random_x = victim.position.x + Enum.random(-game_config.power_ups.distance_to_power_up..game_config.power_ups.distance_to_power_up)
+    random_y = victim.position.y + Enum.random(-game_config.power_ups.distance_to_power_up..game_config.power_ups.distance_to_power_up)
+
+    random_position = %{x: random_x, y: random_y}
+
+    power_up =
+      Entities.new_power_up(
+        last_id,
+        random_position,
+        victim.direction,
+        victim.id
+      )
+
+    put_in(state, [:game_state, :power_ups, last_id], power_up)
+    |> put_in([:game_state, :last_id], last_id)
+    |> spawn_power_ups(victim, amount - 1)
```

Comment:
Why not `Enum.reduce`?

Reply:
This is the best way i found to not spawn power ups when the calculation method returns 0, i though about using a reduce with a range 0..amount but you'll always have at least one cicle with 0. Do you know any way better than this?

Reply:
That worked! i't more readable now thanks